File: src/main/java/aurocosh/divinefavor/common/custom_data/player/data/presence/industrious/IndustriousPresenceDataSerializer.java
package aurocosh.divinefavor.common.custom_data.player.data.presence.industrious;

import aurocosh.divinefavor.common.lib.interfaces.INbtSerializer;
import net.minecraft.nbt.NBTTagCompound;

// Handles the actual read/write of the nbt.
public class IndustriousPresenceDataSerializer implements INbtSerializer<IndustriousPresenceData> {
    private static final String TAG_ORE_TO_BREAK = "OreToBreak";

    @Override
    public void serialize(NBTTagCompound nbt, IndustriousPresenceData instance) {
        nbt.setInteger(TAG_ORE_TO_BREAK, instance.getCount());
    }

    @Override
    public void deserialize(NBTTagCompound nbt, IndustriousPresenceData instance) {
        instance.setCount(nbt.getInteger(TAG_ORE_TO_BREAK));
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/custom_data/player/data/presence/industrious/IndustriousPresenceDataSerializer.kt
package aurocosh.divinefavor.common.custom_data.player.data.presence.industrious

import aurocosh.divinefavor.common.lib.interfaces.INbtSerializer
import net.minecraft.nbt.NBTTagCompound

// Handles the actual read/write of the nbt.
class IndustriousPresenceDataSerializer : INbtSerializer<IndustriousPresenceData> {

    override fun serialize(nbt: NBTTagCompound, instance: IndustriousPresenceData) {
        nbt.setInteger(TAG_ORE_TO_BREAK, instance.count)
    }

    override fun deserialize(nbt: NBTTagCompound, instance: IndustriousPresenceData) {
        instance.count = nbt.getInteger(TAG_ORE_TO_BREAK)
    }

    companion object {
        private const val TAG_ORE_TO_BREAK = "OreToBreak"
    }
}
